src/array/ops/image.rs

```diff
@@ -0,0 +1,336 @@
+use std::borrow::Cow;
+use std::vec;
+
+use image::ImageBuffer;
+
+use crate::datatypes::logical::ImageArray;
+use crate::datatypes::{DataType, Field, ImageMode};
+use crate::error::DaftResult;
+use image::{Luma, LumaA, Rgb, Rgba};
+
+use super::as_arrow::AsArrow;
+use num_traits::FromPrimitive;
+
+use std::ops::Deref;
+
+#[allow(clippy::upper_case_acronyms, dead_code)]
+#[derive(Debug)]
+enum DaftImageBuffer<'a> {
+    L(ImageBuffer<Luma<u8>, Cow<'a, [u8]>>),
+    LA(ImageBuffer<LumaA<u8>, Cow<'a, [u8]>>),
+    RGB(ImageBuffer<Rgb<u8>, Cow<'a, [u8]>>),
+    RGBA(ImageBuffer<Rgba<u8>, Cow<'a, [u8]>>),
+    L16(ImageBuffer<Luma<u16>, Cow<'a, [u16]>>),
+    LA16(ImageBuffer<LumaA<u16>, Cow<'a, [u16]>>),
+    RGB16(ImageBuffer<Rgb<u16>, Cow<'a, [u16]>>),
+    RGBA16(ImageBuffer<Rgba<u16>, Cow<'a, [u16]>>),
+    RGB32F(ImageBuffer<Rgb<f32>, Cow<'a, [f32]>>),
+    RGBA32F(ImageBuffer<Rgba<f32>, Cow<'a, [f32]>>),
+}
+
+macro_rules! with_method_on_image_buffer {
+    (
+    $key_type:expr, $method: ident
+) => {{
+        use DaftImageBuffer::*;
+
+        match $key_type {
+            L(img) => img.$method(),
+            LA(img) => img.$method(),
+            RGB(img) => img.$method(),
+            RGBA(img) => img.$method(),
+            L16(img) => img.$method(),
+            LA16(img) => img.$method(),
+            RGB16(img) => img.$method(),
+            RGBA16(img) => img.$method(),
+            RGB32F(img) => img.$method(),
+            RGBA32F(img) => img.$method(),
+        }
+    }};
+}
+
+impl<'a> DaftImageBuffer<'a> {
+    pub fn height(&self) -> u32 {
+        with_method_on_image_buffer!(self, height)
+    }
+
+    pub fn width(&self) -> u32 {
+        with_method_on_image_buffer!(self, width)
+    }
```

Comment:
Instead of having this macro, would it be possible to implement the `Deref` trait for this enum for `DynamicImage`, since it exposes `height()` and `width()` along with other useful methods? Something like:
```rust
use std::ops::Deref;

impl<'a> Deref for DaftImageBuffer<'a> {
    type Target = DynamicImage;

    fn deref(&self) -> &Self::Target {
        use DaftImageBuffer::*;

        match self {
            L(img) => img.into(),
            LA(img) => img.into(),
            RGB(img) => img.into(),
            RGBA(img) => img.into(),
            L16(img) => img.into(),
            LA16(img) => img.into(),
            RGB16(img) => img.into(),
            RGBA16(img) => img.into(),
            RGB32F(img) => img.into(),
            RGBA32F(img) => img.into(),
        };
    }
}
```